services/provider/server/server.go

```diff
@@ -479,61 +805,63 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 			}),
 		})
 	}
+	return nil
+}
 
+func (s *OCSProviderServer) appendNoobaaExternalResource(
+	ctx context.Context,
+	consumer *ocsv1alpha1.StorageConsumer,
+	extR []*pb.ExternalResource,
+) error {
+	// Noobaa Configuration
 	// Fetch noobaa remote secret and management address and append to extResources
-	consumerName := consumerResource.Name
 	noobaaOperatorSecret := &v1.Secret{}
-	noobaaOperatorSecret.Name = fmt.Sprintf("noobaa-account-%s", consumerName)
+	noobaaOperatorSecret.Name = fmt.Sprintf("noobaa-account-%s", consumer.Name)
 	noobaaOperatorSecret.Namespace = s.namespace
 
-	if err := s.client.Get(ctx, client.ObjectKeyFromObject(noobaaOperatorSecret), noobaaOperatorSecret); err != nil {
-		if kerrors.IsNotFound(err) {
-			// ignoring because it is a provider cluster and the noobaa secret does not exist
-			return extR, nil
-
+	if err := s.client.Get(ctx, client.ObjectKeyFromObject(noobaaOperatorSecret), noobaaOperatorSecret); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to get %s secret. %v", noobaaOperatorSecret.Name, err)
+	} else if !kerrors.IsNotFound(err) {
```

Comment:
There is no need in moving the entire body into an else
```suggestion
	if err := s.client.Get(ctx, client.ObjectKeyFromObject(noobaaOperatorSecret), noobaaOperatorSecret); kerrors.IsNotFound(err) != nil {
		return nil
} else err != nil {
		return fmt.Errorf("failed to get %s secret. %v", noobaaOperatorSecret.Name, err)
	} 
```